routes/user.js

```diff
@@ -203,6 +202,59 @@ router.get('/:id/invitations/in', auth, async (req, res) => {
   }
 });
 
+//GET all user incoming PENDING invitations
+router.get('/:id/invitations/pending', async (req, res) => {
+  const userId = req.params.id;
+  try {
+    const user = await User.findById(userId);
+
+    if (!user) {
+      res.status(404).send('User not found');
+    }
+
+    const pendingInvitesIn = await Invitation.find({
+      toEmail: user.email,
+      status: 'pending',
+    });
+
+    //ATTEMPTING TO DISPLAY THE INVITER BY EMAIL
+    //need to find each user by ID and then retrieve their email
+    // for (let invite of pendingInvitesIn) {
+    //   const user = await User.findById(invite.referrer);
+
+    //   let newInvite = {
+    //     status: invite.status,
+    //     _id: invite._id,
+    //     referrer: invite.referrer,
+    //     createdAt: invite.createdAt,
+    //     referrerEmail: user.email,
+    //   };
+
+    //   pendingInvitesIn.splice(pendingInvitesIn.indexOf(invite, 1, newInvite));
```

Comment:
Can you just do `invite.referrerEmail = user.email`?

Reply:
Yes, but our invite model does not have a key of referrerEmail.  Should I add this to the model?

Reply:
It might be redundant for the base model to have both id and email fro the referrer. Maybe we can make a toObject function like we have on user that does this?

Reply:
I believe you mean something akin to this: 

```
 {
    toJSON: {
      transform: function (doc, ret) {},
    },
    toObject: {
      transform: function (doc, ret) {},
    },
  },
```

Just quite unsure as to how I can get the referrerEmail onto the model this way.

Reply:
So I was thinking like:
```
toObject: {
  transform: function (doc, ret) {
  const user = await User.findById(invite.referrer);
  ret.referrerEmail = user.email;
  return ret;
  },
}
```
though I'm not sure that will work

Reply:
ahh I see.  I wasn't sure of the capabilities of these methods like toObject and toJson but looks like there's a lot they can do!

Reply:
I _think_ they can do anything you want. But I haven't actually used them before, so you should test that out and see if it actually works :smile: